python/lsst/summit/utils/guiders/seeing.py

```diff
@@ -0,0 +1,517 @@
+            ``4e3``.
+
+        Raises
+        ------
+        ValueError
+            If no rows with the given ``expid`` are found in ``stars``.
+        ValueError
+            If more than one unique filter is present in the selected
+            exposure.
+        """
+        self.expid: int = expid
+        self.fluxCut: float = fluxCut
+
+        # Early exit if no data
+        mask = stars["expid"].eq(expid)
+        if not mask.any():
+            raise ValueError(f"Stars dataframe has no data with expid={expid}")
+
+        starsFiltered = stars[mask].copy()
+
+        # Wavelength information from LSST bandpass centers (nm).
+        lamMap: dict[str, float] = {
+            "u": 372.4,
+            "g": 480.7,
+            "r": 622.1,
+            "i": 755.9,
+            "z": 868.0,
+            "y": 975.3,
+        }
+        band = starsFiltered["filter"].unique()
+        if len(band) != 1:
+            raise ValueError("cannot have more than one filter per exposure")
+        self.wavelength: float = float(lamMap[str(band[0])])
+
+        # Initialize metrics builder and detrend stars.
+        # These classes/objects are external to this module.
+        dummyNmissingStampPlaceholder = 0
+        metricsBuilder: GuiderMetricsBuilder = GuiderMetricsBuilder(
+            starsFiltered, dummyNmissingStampPlaceholder
+        )
+        self.metrics: pd.DataFrame = metricsBuilder.buildMetrics(self.expid)
+        starsDetrended: pd.DataFrame = metricsBuilder.detrendStars()
+
+        # Build star list per detector with a flux threshold.
+        self.starList: dict[str, pd.DataFrame] = groupStarsPerDetector(starsDetrended, fluxCut)
+        self.detectors: list[str] = list(self.starList.keys())
+
+    def measureVar(self) -> list[float]:
+        """
+        Compute per-detector variances of corrected centroid motions.
+
+        Requires that detrending has already been applied. Variances are
+        scaled to units of (pix × 100)^2, i.e., (centipix)^2.
+
+        Returns
+        -------
+        out : `list[float]`
+            Flattened list of per-detector variances:
+            ``[vx_det1, vy_det1, vx_det2, vy_det2, ...]``. If a detector
+            was excluded or lacks corrected coordinates, entries are
+            ``NaN``.
+        """
+        out: list[float] = []
+        for _, df in self.starList.items():
+            meanFlux = float(np.nanmean(pd.to_numeric(df["flux"], errors="coerce")))
+            hasCorr = "dxfp_corr" in df and "dyfp_corr" in df
+            if meanFlux > self.fluxCut and hasCorr:
+                vx = float(np.nanvar(pd.to_numeric(df["dxfp_corr"], errors="coerce"))) * (100.0**2)
+                vy = float(np.nanvar(pd.to_numeric(df["dyfp_corr"], errors="coerce"))) * (100.0**2)
+                out.extend([vx, vy])
+            else:
+                out.extend([np.nan, np.nan])
+        return out
+
+    @staticmethod
+    def makePairs(kind: str) -> list[list[str]]:
```

Comment:
Minor point, but I'd call this `getPairs` rather than "make" as nothing is really being _made_, as such. Also, I tend to prefer longer names, like `getDetectorNamePairs()` or something like that, so it's clear what's going on in the calling code, but I'm quite extreme like that, and others care less. I'd also recommend returning a tuple of tuples, so that these are immutable, to save bugs downstream.